src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -138,31 +143,81 @@ protected TupleBatch fetchNextReady() throws IOException, DbException {
         throw new DbException("Error parsing row " + lineNumber, e);
       }
       CSVRecord record = iterator.next();
+      // If the number of columns matches the schema for the first line, mark as discarded
+      if (record.size() == schema.numColumns() && lineNumber - 1 == 0 && partitionStartByteRange != 0) {
+        // Check if the previous 2 bytes contains a new line -- this should be a rare case
+        InputStreamReader startStreamReader =
+            new InputStreamReader(source.getInputStream(partitionStartByteRange - 1, partitionStartByteRange));
+        char currentChar = (char) startStreamReader.read();
+        LOGGER.warn("CHAR BEFORE " + currentChar);
+        if (currentChar != '\n') {
+          discardedRecord = true;
+        } else {
+          LOGGER.warn("CHAR MATCH");
+        }
+      }
+      // If the number of columns matches the schema for the last line, make sure to read entire line
+      else if (record.size() == schema.numColumns() && !iterator.hasNext() && !isLastWorker) {
+        long movingEndByte = partitionEndByteRange + MyriaConstants.BYTE_OVERLAP_PARALLEL_INGEST;
+        long byteAtBeginningOfRecord = record.getCharacterPosition();
+        boolean newLineFound = false;
+
+        while (!newLineFound) {
+          // Keep going back
+          movingEndByte += MyriaConstants.BYTE_OVERLAP_PARALLEL_INGEST;
+          trailingEndInputStream = source.getInputStream(partitionEndByteRange, movingEndByte);
+
+          InputStreamReader startStreamReader = new InputStreamReader(trailingEndInputStream);
+          int dataChar = startStreamReader.read();
+          while (dataChar != -1) {
+            char currentChar = (char) dataChar;
+            if (currentChar == '\n') {
```

Comment:
To be consistent with `CSVFormat`'s documented behavior, we need to consider `\r`, `\n`, and `\r\n` as valid record delimiters.